lib/react_client.dart

```diff
@@ -146,6 +150,73 @@ class ReactDartComponentFactoryProxy<TComponent extends Component>
   }
 }
 
+/// Creates ReactJS [Component2] instances for Dart components.
+class ReactDartComponentFactoryProxy2<TComponent extends Component2>
+    extends ReactComponentFactoryProxy
+    implements ReactDartComponentFactoryProxy {
+  /// The ReactJS class used as the type for all [ReactElement]s built by
+  /// this factory.
+  final ReactClass reactClass;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  final ReactJsComponentFactory reactComponentFactory;
+
+  final Map defaultProps;
+
+  ReactDartComponentFactoryProxy2(ReactClass reactClass)
+      : this.reactClass = reactClass,
+        this.reactComponentFactory = React.createFactory(reactClass),
+        this.defaultProps = new JsBackedMap.fromJs(reactClass.defaultProps);
+
+  ReactClass get type => reactClass;
+
+  ReactElement build(Map props, [List childrenArgs = const []]) {
+    // TODO if we don't pass in a list into React, we don't get a list back in Dart...
+
+    List children;
+    if (childrenArgs.isEmpty) {
+      children = childrenArgs;
+    } else if (childrenArgs.length == 1) {
+      final singleChild = listifyChildren(childrenArgs[0]);
+      if (singleChild is List) {
+        children = singleChild;
+      }
+    }
+
+    if (children == null) {
+      // FIXME are we cool to modify this list?
+      // FIXME why are there unmodifiable lists here?
+      children = childrenArgs.map(listifyChildren).toList();
+      markChildrenValidated(children);
+    }
+
+    return reactComponentFactory(
+      generateExtendedJsProps(props),
+      children,
+    );
+  }
+
+  /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
+  /// consumption by the [react] library internals.
+  static JsMap generateExtendedJsProps(Map props) {
+    final propsForJs = new JsBackedMap.from(props);
+
+    // FIXME forwarded DOM props???
```

Comment:
Would DOM props not be forwarded? seems like this works without any intervention...